### PR TITLE
Pubber endpoint configuration working with mosquitto broker

### PR DESCRIPTION
**Pull request by grafnu**

### CODE REVIEW

**Review by noursaidi — APPROVED** [review]

File: bin/test_mosquitto [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        }
+      }
+    },
+    "sitePath": "$site_path",
+    "deviceId": "$device_id",
+    "serialNo": "$serial_no"
+}
+EOF
+
+ps ax | fgrep java | fgrep /tmp/pubber_config.json | fgrep -v grep | awk '{print $1}' | xargs -r kill
```
Comment (johnrandolph) [inline bin/test_mosquitto:67]: something like I'm doing eleswhere?  

function kill_existing() {
  kill $(pgrep -f 'java.*pubber-.*jar')
}

File: tests/configuration_endpoint.tests/multiple.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "protocol": "mqtt",
+  "hostname": "mqtt.googleapis.com",
+  "port": 8883,
+  "client_id": "prasjdhsadas",
```
Comment (johnrandolph) [inline tests/configuration_endpoint.tests/multiple.json:5]: What is this - just a placeholder?

File: bin/test_mosquitto [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+USERNAME=scrumptus
+PASSWORD=aardvark
+
+echo password_file ${PASSWORD_FILE} | sudo tee /etc/mosquitto/conf.d/udmi_test.conf
+sudo sed -i 's/allow_anonymous true/allow_anonymous false/' /etc/mosquitto/mosquitto.conf
+sudo sed -i 's/#listener/listener/' /etc/mosquitto/mosquitto.conf
+
+sudo touch ${PASSWORD_FILE}
+sudo mosquitto_passwd -b ${PASSWORD_FILE} ${USERNAME} ${PASSWORD}
+sudo chmod 666 ${PASSWORD_FILE}
```
Comment (johnrandolph) [inline bin/test_mosquitto:30]: Eventually we can't do this kind of stuff

**Review by johnrandolph — APPROVED** [review]

File: bin/test_mosquitto (hunk 3 above)
Comment (grafnu) [reply bin/test_mosquitto:30]: Which kind of stuff, exactly?

File: tests/configuration_endpoint.tests/multiple.json (hunk 2 above)
Comment (grafnu) [reply tests/configuration_endpoint.tests/multiple.json:5]: The file is just for testing against various kinds of schema violations.... not even sure that it counts as a placeholder.

File: bin/test_mosquitto (hunk 1 above)
Comment (grafnu) [reply bin/test_mosquitto:68]: done.

**Pull request merged**